jaclang_jaseci/plugins/common.py

```diff
@@ -0,0 +1,1174 @@
+        default: Optional[Union["NodeArchitype", "EdgeArchitype"]] = None,
+    ) -> Optional[Union["NodeArchitype", "EdgeArchitype"]]:
+        """Retrieve Architype in memory."""
+        return self.__mem__.get(ObjectId(id), default)
+
+    def set(
+        self, id: Union[ObjectId, str], obj: Union[NodeArchitype, EdgeArchitype]
+    ) -> None:
+        """Push Architype in memory via ID."""
+        self.__mem__[ObjectId(id)] = obj
+
+    def remove(
+        self, id: Union[ObjectId, str]
+    ) -> Optional[Union["NodeArchitype", "EdgeArchitype"]]:
+        """Pull Architype in memory via ID."""
+        return self.__mem__.pop(ObjectId(id), None)
+
+    def report(self, obj: Any) -> None:  # noqa: ANN401
+        """Append report."""
+        self.reports.append(obj)
+
+    def response(self, returns: list[Any], status: int = 200) -> dict[str, Any]:
+        """Return serialized version of reports."""
+        resp = {"status": status, "returns": returns}
+
+        if self.reports:
+            for key, val in enumerate(self.reports):
+                if isinstance(val, DocArchitype) and (
+                    ret_jd := getattr(val, "_jac_doc_", None)
+                ):
+                    self.reports[key] = {"id": ret_jd.ref_id, **val.get_context()}
+                else:
+                    self.clean_response(key, val, self.reports)
+            resp["reports"] = self.reports
+
+        return resp
+
+    def clean_response(
+        self, key: Union[str, int], val: Any, obj: Union[list, dict]  # noqa: ANN401
+    ) -> None:
+        """Cleanup and override current object."""
+        if isinstance(val, list):
+            for idx, lval in enumerate(val):
+                self.clean_response(idx, lval, val)
+        elif isinstance(val, dict):
+            for key, dval in val.items():
+                self.clean_response(key, dval, val)
+        elif isinstance(val, DocArchitype):
+            cast(dict, obj)[key] = {"id": val._jac_doc_.ref_id, **val.get_context()}
+
+
+async def async_filter(
+    edges: list[Union["EdgeArchitype", DocAnchor["EdgeArchitype"]]]
+) -> AsyncGenerator[tuple[EdgeArchitype, NodeArchitype, NodeArchitype], None]:
+    """Async filter for edges."""
+    for edge in edges:
+        if (
+            (ed := await edge.connect())
+            and (edj := ed._jac_)
+            and (edjs := edj.source)
+            and (edjt := edj.target)
+            and (src := await edjs.connect())
+            and (tgt := await edjt.connect())
+        ):
+            yield (ed, src, tgt)
+
+
+Root.__name__ = ""
+GenericEdge.__name__ = ""
+
+JCLASS: dict[str, dict[str, type]] = {"n": {"": Root}, "e": {"": GenericEdge}}
+
+base_root_id = ObjectId("000000000000000000000000")
+base_root = Root.Collection.build_node(
+    DocAnchor[Root](
+        type=JType.node,
+        id=base_root_id,
+        root=base_root_id,
+        access=DocAccess(),
+        connected=True,
+    ),
+    {"context": {}, "edge": []},
+)
```

Comment:
Are these (after line 1159) hardcoded to create root node? shouldn't be done per user and upon user creation?

Reply:
This is for Generic Root instance used for non authenticated users